### PR TITLE
c8d: Fix pull progress update

### PR DESCRIPTION
**Pull request by rumpl**
**- What I did**

We were sending the "Pulling from ..." message too early, if the pull progress wasn't able to resolve the image we wouldn't sent the error back. Sending that first message would have flushed the output stream and image_routes.go would return a nil error.

So instead of sending the "Pulling from ..." message right away, we wait for the pull progress to at least start pulling a manifest, which is kinda what the graph driver implementation does.

**- How I did it**

**- How to verify it**

`TestDaemonProxy` should not fail any more

```console
$ make DOCKER_GRAPHDRIVER=overlayfs TEST_FILTER=TestDaemonProxy TEST_INTEGRATION_USE_SNAPSHOTTER=1 TEST_IGNORE_CGROUP_CHECK=1 test-integration
```

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: daemon/containerd/image_pull.go [hunk 1]
```diff
@@ -73,8 +73,14 @@ func (i *ImageService) PullImage(ctx context.Context, image, tagOrDigest string,
 	finishProgress := jobs.showProgress(ctx, out, pp)
 	defer finishProgress()
 
+	var sentPullingFrom bool
 	ah := images.HandlerFunc(func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
 		if images.IsManifestType(desc.MediaType) {
+			if !sentPullingFrom {
```
Comment (thaJeztah) [inline daemon/containerd/image_pull.go:79]: would this code always be executed (also if the manifest happens to be cached already)?

Comment (rumpl) [reply daemon/containerd/image_pull.go:79]: Yup

Comment (thaJeztah) [reply daemon/containerd/image_pull.go:79]: Thanks for checking!

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by vvoland — APPROVED** [review]

File: daemon/containerd/image_pull.go (hunk 1 above)
Comment (vvoland) [reply daemon/containerd/image_pull.go:79]: Fixed the unfinished downloading message: https://github.com/moby/moby/pull/46515

**Pull request merged**